imap/mboxlist.c

```diff
@@ -5331,6 +5331,11 @@ EXPORTED int mboxlist_changesub(const char *name, const char *userid,
         mboxevent_free(&mboxevent);
     }
 
+    xsyslog(LOG_NOTICE, "auditlog: %s",
+                        "sessionid=<%s> userid=<%s> mailbox=<%s>",
+            (add ? "subscribe" : "unsubscribe"),
+            session_id(), userid, name);
+
```

Comment:
This won't work, the first argument to xsyslog is a simple event description, not a printf format string.  This is going to end up logging something like:

```
auditlog: %s: sessionid=<subscribe> userid=<[session id]> mailbox=<[userid]> 
```

It might be nice to add an auditlog() function specifically for audit logging, but for now we don't have one, just use regular syslog.

Reply:
I just did the simplest possible dumb thing (split it into two separate xsyslogs, one for each case)